### PR TITLE
BREAKING CHANGE: showcase support

### PR DESCRIPTION
**Pull request by L0wry**
find showcase files and filters out non story files. 
Creates a dextrose.tmp file to be consumed by RNSL

### CODE REVIEW

File: bin/run.js [hunk 1]
```diff
@@ -88,15 +88,30 @@ program
...
 
+
+program
+.command('generate-showcase-files [dirPath]')
+.alias('gsf')
+.action((dirPath) => {
+  if (!dirPath) log.error('run', 'no directory path found ');
+
+  log.info('run', 'Generating Dextrose showcase files');
+  generateStories(dirPath)
```
Comment (milesillsley) [inline bin/run.js]: I think this needs to be generateShowcaseFiles

File: bin/generate-showcase-files.js [hunk 2]
```diff
@@ -0,0 +1,67 @@
+const fs = require('fs');
+const path = require('path');
+const pretty = require('js-object-pretty-print').pretty
```
Comment (milesillsley) [inline bin/generate-showcase-files.js]: Does js-object-pretty-print need to be included in the package.json?

File: bin/generate-showcase-files.js [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    files.forEach((file) => {
+      const showcaseObject = require(path.resolve(dirPath, file)).default
+      showcaseObject.children = showcaseObject.children.filter(currentValue => currentValue.type.includes('story')); 
+      
+      fs.readFile(file, 'utf8', (err, data) => {
+        if (err) throw err;
+
+        const showcaseComponentImports = data.split('export')[0];
+        const indentSpacing = 4;
+        const showCaseCompoentObjectString = pretty(showcaseObject, indentSpacing, 'PRINT', true);
```
Comment (shendriksen) [inline bin/generate-showcase-files.js]: small typo in 'component' here. Maybe it called be called 'showcaseComponentAsString' if that's what it's doing?

File: bin/generate-showcase-files.js [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+        const fileToWrite = `${showcaseComponentImports}export default  ${showCaseCompoentObjectString}`;
+
+        const fileName = file.split('.js')[0];
```
Comment (shendriksen) [inline bin/generate-showcase-files.js]: Might be worth creating a file utility class to abstract some of the implementation away from here. You could end up with 'file.name', 'file.extension' etc then, which would read nicer.

File: bin/generate-showcase-files.js [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+      }
+    });
+  });
+
+const generateShowcaseFiles = dir => new Promise((resolve, reject) => {
+  findShowcaseFiles(dir)
+    .then((showcaseFiles) => {
+      createDextroseTmp(dir, showcaseFiles);
+
+      resolve();
```
Comment (shendriksen) [inline bin/generate-showcase-files.js]: You shouldn't need this call as your createDextroseTmp returns a promise.

**Pull request merged**